rotkehlchen/chain/ethereum/modules/eth2/decoder.py

```diff
@@ -65,23 +66,27 @@ def _decode_eth2_deposit_event(self, context: DecoderContext) -> DecodingOutput:
                 event.event_type == HistoryEventType.SPEND and
                 event.event_subtype == HistoryEventSubType.NONE and
                 event.asset == A_ETH and
-                event.balance.amount == amount
+                event.balance.amount >= amount
             ):
-                replace_event_idx = idx
-                break
-        else:  # loop did not encounter a break - index not found
-            log.error(f'While decoding ETH deposit event {context.transaction.tx_hash.hex()} for public key {public_key} could not find the send event')  # noqa: E501
-            return DEFAULT_DECODING_OUTPUT
-
-        old_event = context.decoded_events[replace_event_idx]
-        context.decoded_events[replace_event_idx] = EthDepositEvent(
-            tx_hash=context.transaction.tx_hash,
-            validator_index=validator_index,
-            sequence_index=old_event.sequence_index,
-            timestamp=old_event.timestamp,
-            balance=old_event.balance,
-            depositor=old_event.location_label,  # type: ignore  # it's an address
-            extra_data=extra_data,  # only used if validator index is unknown
+                eth_deposit_event = EthDepositEvent(
+                    tx_hash=context.transaction.tx_hash,
+                    validator_index=validator_index,
+                    sequence_index=event.sequence_index,
+                    timestamp=event.timestamp,
+                    balance=Balance(amount=amount),
+                    depositor=event.location_label,  # type: ignore  # it's an address
```

Comment:
I would suggest against using the type ignores. I would use `string_to_evm_address` and add to the condition that `location_label is not None` or do an assertion

Reply:
This is the code written by Lefteris, but yeah, I can implement the suggestion